python/ray/autoscaler/_private/kuberay/utils.py

```diff
@@ -1,6 +1,17 @@
 # Source:
 # https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py
 from decimal import Decimal, InvalidOperation
+from functools import reduce
+from typing import Optional
+
+# Mapping used to get generation for TPU-{accelerator}-head resource
+# https://cloud.google.com/kubernetes-engine/docs/how-to/tpus#run
+gke_tpu_accelerator_to_generation = {
```

Comment:
Is this only for GKE, or does GCE also have the same thing?

Reply:
The association between the TPU accelerator name and the TPU version (v4, v5e, v5p, etc.) is the same for both GKE and GCE.

Reply:
Is it possible to unify the code path of the association logic if it is the same in both the VM launcher and KubeRay?

Reply:
I think I misspoke a bit, GCE provisions TPUs through individual VMs while GKE provisions TPU podslices using node pools. GKE uses `gke-accelerator-type` and `gke-tpu-topology` nodeSelectors to schedule workloads on the correct TPU nodes. In terms of how we generate the TPU head resource, while the end result is the same, the accelerator type we see for GCE will look like "v3", "v4", etc. while for GKE it's of the form "tpu-v4-podslice". Thus, there's really no code path to unify here, since GCE and GKE generate the head resource differently.